### PR TITLE
MONGOID-5322: Remove deprecated geo_spacial method (was aliased to geo_spatial)

### PR DESCRIPTION
**Pull request by johnnyshields**
Includes release notes.

### CODE REVIEW

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/MONGOID-5322

**Comment by Neilshweky** [comment]
@johnnyshields please rebase this on master when you get the chance

File: spec/mongoid/criteria/queryable/selectable_spec.rb [hunk 1]
```diff
@@ -695,127 +695,119 @@ def localized?
     end
   end
 
-  %i(geo_spatial geo_spacial).each do |meth|
-    describe "#geo_spacial" do
+  describe "#geo_spatial" do
 
-      let(:query_method) { meth }
+    let(:query_method) { :geo_spatial }
```
[the remark this answers is not in the document]
Comment (johnnyshields) [reply spec/mongoid/criteria/queryable/selectable_spec.rb:700]: Yes, it is used by some of the `it_behaves_like` blocks below.

**Comment by johnnyshields** [comment]
@Neilshweky rebased and responded to comment.

**Review by p-mongo — APPROVED** [review]

**Pull request merged**